### PR TITLE
tracee-ebpf: add security_path_symlink event

### PR DESCRIPTION
**Pull request by OriGlassman**

### CODE REVIEW

**Review by AlonZivony — APPROVED** [review]
Seems good to me.
@yanivagman I think you can approve this quickly for @OriGlassman

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 1]
```diff
@@ -3035,6 +3036,29 @@ int BPF_KPROBE(trace_security_socket_create)
     return events_perf_submit(&data, SECURITY_SOCKET_CREATE, 0);
 }
 
+SEC("kprobe/security_path_symlink")
```
Comment (yanivagman) [inline tracee-ebpf/tracee/tracee.bpf.c:3039]: security_path_* is guarded with CONFIG_SECURITY_PATH:
https://elixir.bootlin.com/linux/latest/source/include/linux/security.h#L1767
This config is not enabled on some distros, so we better not use it, and definitely not as a default event, which will fail tracee to start.

security_inode_symlink can be used instead

**Review by yanivagman — CHANGES_REQUESTED** [review]

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -187,9 +187,10 @@ Copyright (C) Aqua Security inc.
 #define SECURITY_KERNEL_READ_FILE       1028
 #define SECURITY_INODE_MKNOD            1029
 #define SECURITY_POST_READ_FILE         1030
-#define SOCKET_DUP                      1031
-#define HIDDEN_INODES                   1032
-#define MAX_EVENT_ID                    1033
+#define SECURITY_INODE_SYMLINK           1031
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:190]: please fix number indentation

File: pkg/ebpf/events_definitions.go [hunk 3]
```diff
@@ -6082,6 +6083,18 @@ var EventsDefinitions = map[int32]EventDefinition{
 			{Type: "dev_t", Name: "dev"},
 		},
 	},
+	SecurityInodeSymlinkEventId: {
+		ID32Bit: sys32undefined,
+		Name:    "security_inode_symlink",
+		Probes: []probe{
+			{event: "security_inode_symlink", attach: kprobe, fn: "trace_security_inode_symlink"},
+		},
+		Sets: []string{"default", "lsm_hooks", "fs", "fs_file_ops"},
```
Comment (yanivagman) [inline pkg/ebpf/events_definitions.go:6092]: If you don't mind, let's remove this from the default set for now

File: tracee-ebpf/tracee/tracee.bpf.c (hunk 1 above)
Comment (yanivagman) [reply tracee-ebpf/tracee/tracee.bpf.c]: > Fixed - There is one downside now though: 2 events are generated when a 'ln -s' is happening inside a container. One with the paths relevant for the container, and the other for the overlay fs. Unlike for when hooking security_path_symlink, which happens only in the container scope.

There might be an easy fix for this.
By looking at inode->i_sb->s_magic and comparing it to overlayfs magic number, you can know if the path is within the container or not

**Review by yanivagman — APPROVED** [review]

**Pull request merged**